app/ide-desktop/lib/client/src/bin/server.ts

```diff
@@ -22,14 +24,11 @@ const HTTP_STATUS_OK = 200
 // === URL Parameters ===
 // ======================
 
-/** Construct URL query with the given parameters. For each `key` - `value` pair,
+/** Construct a URL query string with the given parameters. For each `key` - `value` pair,
  * `key=value` will be added to the query. */
 export function urlParamsFromObject(obj: Record<string, string>) {
-    const params = []
-    for (const [key, value] of Object.entries(obj)) {
-        params.push(`${key}=${encodeURIComponent(value)}`)
-    }
-    return params.length === 0 ? '' : '?' + params.join('&')
+    const params = new URLSearchParams(obj).toString()
+    return params ? '?' + params : ''
```

Comment:
This is a really neat change, yet at the same time, it got me thinking, about whether we need this function at all. It being `string`-ly typed is a code smell to me.
Perhaps we can construct URL by using the URL API rather than by pretty-printing pieces of text together.

Reply:
unfortunately it seems even the url api requires properties to be `string`s... but i think it might be a good idea to completely remove `urlParamsFromObject` anyway as the URL APIs can do the same thing?